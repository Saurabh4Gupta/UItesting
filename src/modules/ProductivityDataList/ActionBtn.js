import React from 'react';
import { Menu } from '@dentsu-ui/components';
import PropTypes from 'prop-types';

const ActionBtn = (props) => {
  const {
    handleToggleData,
    actionName,
    deleteBtn,
    showStatus,
    handleDeleteModel,
  } = props;
  return (
    <>
      <Menu>
        <Menu.Button />
        <Menu.List>
          <Menu.Item onClick={handleToggleData}>{actionName}</Menu.Item>
          {showStatus === false ? (
            ''
          ) : (
            <Menu.Item onClick={handleDeleteModel}>
              {deleteBtn}
            </Menu.Item>
            )}
        </Menu.List>
      </Menu>
    </>
  );
};
ActionBtn.propTypes = {
  handleToggleData: PropTypes.func.isRequired,
  actionName: PropTypes.string.isRequired,
  deleteBtn: PropTypes.string.isRequired,
  showStatus: PropTypes.string.isRequired,
  handleDeleteModel: PropTypes.func,
};
ActionBtn.defaultProps = {
  handleDeleteModel: true,
};

export default ActionBtn;
