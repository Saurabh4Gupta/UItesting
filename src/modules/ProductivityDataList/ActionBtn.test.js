import React from 'react';
import { shallow } from 'enzyme';
import  ActionBtn from './ActionBtn';

describe('<EmptyState />', () => {
  let wrapper;
  const props = {
    handleToggleData: jest.fn(),
    actionName:'test1',
    deleteBtn:'test2',
    showStatus:true,
    objId: 123,
    handleDeleteModel: jest.fn(),
  }
  beforeAll(() => {
    wrapper = shallow(<ActionBtn {...props} />);
  })
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('should render the else part', () => {
    wrapper.find('MenuItem').at(1).props().onClick();
    expect(wrapper.props().handleDeleteModel.calls.length).toBe(1);
  })
})
