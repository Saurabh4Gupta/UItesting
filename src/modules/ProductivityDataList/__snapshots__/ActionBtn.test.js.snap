// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should match snapshot 1`] = `
<Fragment>
  <Menu>
    <ForwardRef />
    <ForwardRef>
      <MenuItem
        onClick={[MockFunction]}
      >
        test1
      </MenuItem>
      <MenuItem
        onClick={[Function]}
      >
        test2
      </MenuItem>
    </ForwardRef>
  </Menu>
</Fragment>
`;
